torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,479 @@
+
+        image = torch.cat(list_modals, dim=0)
+
+        if self.transforms is not None:
+            sample_trans: dict[str, Tensor] = {"image": image, "mask": mask}
+
+            transformed = self.transforms(sample_trans)
+
+            image = rearrange(transformed["image"], "h w c -> c h w")
+            mask = rearrange(transformed["mask"], "h w c -> c h w")
+
+        sample: dict[str, Tensor] = {"image": image, "mask": mask}
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.ids)
+
+    def _load_raster(self, filename: int, source: str) -> Tensor:
+        """Load a single raster image or target.
+
+        Args:
+            filename: name of the file to load
+            source: the filename of the modality
+
+        Returns:
+            the raster image or target
+        """
+        with rasterio.open(os.path.join(self.root, source, f"{filename}.tif")) as f:
+            array = f.read()
+            if array.dtype == np.uint16:
+                array = array.astype(np.int32)
+            tensor = torch.from_numpy(array)
+            return tensor
+
+    def _verify(self, url: str, md5: Optional[str] = None) -> None:
+        """Verify the integrity of the dataset.
+
+        Args:
+            url: url to the file
+            md5: md5 of the file to be verified
+
+        Raises:
+            RuntimeError: if dataset is not found
+        """
+        modality_folder_name = os.path.split(url)[-1]
+        mod_fold_no_ext = modality_folder_name.split(".")[0]
+        modality_path = os.path.join(self.root, mod_fold_no_ext)
+        split_path = os.path.join(self.root, modality_folder_name)
+        if mod_fold_no_ext == "split_IDs":
+            modality_path = split_path
+
+        # Check if the files already exist
+        if os.path.exists(modality_path):
+            return
+
+        # Check if the zip files have already been downloaded, if so, extract
+        filepath = os.path.join(self.root, url.split("/")[-1])
+        if os.path.isfile(filepath) and filepath.endswith(".zip"):
+            if self.checksum and not check_integrity(filepath, md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            self._extract(url)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                "Dataset not found in `root` directory and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+        elif self.download and url.endswith(".csv"):
+            # Download the split file
+            download_url(url, self.root, filename=os.path.split(url)[1], md5=None)
+
+        elif self.download and not url.endswith(".csv"):
```

Comment:
```suggestion
        elif url.endswith(".csv"):
            # Download the split file
            download_url(url, self.root, filename=os.path.split(url)[1], md5=None)

        else:
```
`self.download` is guaranteed to be true thanks to the first if-block